test/integration/sapi-util.h

```diff
@@ -35,6 +35,16 @@
 #define TSS2_APP_ERROR(base_rc)   (TSS2_APP_ERROR_LEVEL | base_rc)
 #define TSS2_APP_RC_BAD_REFERENCE  TSS2_APP_ERROR (TSS2_BASE_RC_BAD_REFERENCE)
 /*
+ */
+#define TSS2_RETRY_EXP(expression)                         \
+    ({                                                     \
+        TSS2_RC __result = 0;                              \
+        do {                                               \
+            __result = (expression);                       \
+        } while ((__result & 0x0000ffff) == TPM_RC_RETRY); \
```

Comment:
Good question. The only documentation for this RC is in part 2 and part 3 of the TPM spec. In both cases (part 2 table 16, part 3 table 4) the descriptive text is the same:

```
the TPM was not able to start the command
```
Not much to go on. But based on my experimentation with the simulator, the second try always succeeds so I figured this code was sufficient. Not sure it's appropriate but I modeled this after `TEMP_FAILURE_RETRY`. This relies on the `EINTR` errno which is a rare and temporary AFAIK. But again, this may not be appropriate and will largely depend on how TPM manufacturers use the RC, which may not be the same as the way the simulator uses it..

All that said, the patch adds the macro to a header in the test harness. It'll won't be installed on a users systems, it should only be consumed by the test cases / harness and so if one of the test cases ends up stuck in an infinite loop I'm not too concerned. If we move it to a public header I'd say we would want better data on how the RC is being used in the wild.